test/e2e/v1beta1/scripts/gh-actions/setup-katib.sh

```diff
@@ -65,10 +65,24 @@ echo "Deploying Katib"
 cd ../../../../../ && WITH_DATABASE_TYPE=$WITH_DATABASE_TYPE make deploy && cd -
 
 # Wait until all Katib pods is running.
+MAX_RETRY_COUNT=10
+until [ "${MAX_RETRY_COUNT}" = "0" ] || [ "$(kubectl get pods -n kubeflow -l "katib.kubeflow.org/component in ($WITH_DATABASE_TYPE,controller,ui,db-manager)" -oname | wc -l)" = "4" ]; do
+  echo "Waiting for katib components to be created"
+  sleep 2
+  MAX_RETRY_COUNT=$((MAX_RETRY_COUNT - 1))
+done
+
 TIMEOUT=120s
 kubectl wait --for=condition=ready --timeout=${TIMEOUT} -l "katib.kubeflow.org/component in ($WITH_DATABASE_TYPE,controller,db-manager,ui)" -n kubeflow pod ||
   (kubectl get pods -n kubeflow && kubectl describe pods -n kubeflow && exit 1)
 
+MAX_RETRY_COUNT=10
+until [ "${MAX_RETRY_COUNT}" = "0" ] || [ "$(kubectl get pod -n kubeflow -l katib.kubeflow.org/component=controller -ojson | jq '.items[].status.conditions[] | select(.type=="ContainersReady").status')" = "True" ]; do
+  echo "Waiting for Katib controller to be ready"
+  sleep 2
+  MAX_RETRY_COUNT=$((MAX_RETRY_COUNT - 1))
+done
+
```

Comment:
The `kubectl wait` command doesn't wait for pods to be ready:

```shell
/home/runner/work/katib/katib/test/e2e/v1beta1/scripts/gh-actions
pod/katib-db-manager-756ff4d844-wrs27 condition met
All Katib components are running.
Katib deployments
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
katib-controller   0/1     0            0           2s
katib-db-manager   1/1     1            1           2s
katib-mysql        0/1     0            0           2s
```

https://github.com/kubeflow/katib/actions/runs/5842762640/job/15844213190?pr=2201#step:4:5605

Reply:
I changed the way to avoid this flakiness.